src/main.c

```diff
@@ -57,6 +57,96 @@ extern bool g_dpvs_pdump;
+        DPVS_MODULE(MODULE_NETIF_VDEV,  "vdevs",                \
+                    netif_vdevs_add,     NULL),                 \
+        DPVS_MODULE(MODULE_TIMER,       "timer",                \
+                    dpvs_timer_init,     dpvs_timer_term),      \
+        DPVS_MODULE(MODULE_TC,          "tc",                   \
+                    tc_init,             NULL),                 \
+        DPVS_MODULE(MODULE_NETIF,       "netif",                \
+                    netif_init,          netif_term),           \
+        DPVS_MODULE(MODULE_CTRL,        "cp",                   \
+                    ctrl_init,           ctrl_term),            \
+        DPVS_MODULE(MODULE_TC_CTRL,     "tc cp",                \
+                    tc_ctrl_init,        NULL),                 \
+        DPVS_MODULE(MODULE_VLAN,        "vlan",                 \
+                    vlan_init,           NULL),                 \
+        DPVS_MODULE(MODULE_INET,        "inet",                 \
+                    inet_init,           inet_term),            \
+        DPVS_MODULE(MODULE_SA_POOL,     "sa_pool",              \
+                    sa_pool_init,        sa_pool_term),         \
+        DPVS_MODULE(MODULE_IP_TUNNEL,   "tunnel",               \
+                    ip_tunnel_init,      ip_tunnel_term),       \
+        DPVS_MODULE(MODULE_VS,          "ipvs",                 \
+                    dp_vs_init,          dp_vs_term),           \
+        DPVS_MODULE(MODULE_NETIF_CTRL,  "netif ctrl",           \
+                    netif_ctrl_init,     netif_ctrl_term),      \
+        DPVS_MODULE(MODULE_IFTRAF,      "iftraf",               \
+                    iftraf_init,         iftraf_term),          \
+        DPVS_MODULE(MODULE_EAL_MEM,     "iftraf",               \
+                    eal_mem_init,        eal_mem_term),         \
+        DPVS_MODULE(MODULE_LAST,        "last module",          \
+                    NULL,                NULL)                  \
+    }
+
+#define DPVS_MODULE(a, b, c, d)  a
+enum dpvs_modules DPVS_MODULES;
+#undef DPVS_MODULE
+
+#define DPVS_MODULE(a, b, c, d)  b
+static const char *dpvs_modules[] = DPVS_MODULES;
+#undef DPVS_MODULE
+
+typedef int (*dpvs_module_init_pt)(void);
+typedef int (*dpvs_module_term_pt)(void);
+
+#define DPVS_MODULE(a, b, c, d)  c
+dpvs_module_init_pt dpvs_module_inits[] = DPVS_MODULES;
+#undef DPVS_MODULE
+
+#define DPVS_MODULE(a, b, c, d)  d
+dpvs_module_term_pt dpvs_module_terms[] = DPVS_MODULES;
+
+static void modules_init(void)
+{
+    int m, err;
+
+    for (m = MODULE_FIRST; m < MODULE_LAST; m++) {
+        if (dpvs_module_inits[m]) {
+            if ((err = dpvs_module_inits[m]()) != EDPVS_OK) {
+                rte_exit(EXIT_FAILURE, "failed to init %s: %s\n",
+                         dpvs_modules[m], dpvs_strerror(err));
+            }
+        }
+    }
+}
+
+static void modules_term(void)
+{
+    int m, err;
+
+    for (m = MODULE_LAST; m >= MODULE_FIRST; m--) {
```

Comment:
the initial value for m should be `MODULE_LAST - 1`

Reply:
Now the new patch makes "iftraf" as MODULE_LAST.